engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -207,8 +212,24 @@ private void initMaterials() {
     private void initRenderGraph() {
         // FIXME: init pipeline without specifying them as a field in this class
         NodeFactory nodeFactory = new NodeFactory(context);
+
+        // ShadowMap generation
+        FBOConfig shadowMapConfig =
+                new FBOConfig(ShadowMapNode.SHADOW_MAP, FBO.Type.NO_COLOR, shadowMapResolutionDependentFBOs).useDepthBuffer();
+        BufferClearingNode.RequiredData shadowMapClearingData =
+                new BufferClearingNode.RequiredData(shadowMapConfig, GL_DEPTH_BUFFER_BIT);
+        Node shadowMapClearingNode = nodeFactory.createInstance(BufferClearingNode.class, shadowMapClearingData);
```

Comment:
This might just be me (and I'm hardly an architect) but when I think about inner classes it is something exclusively used by the wrapping class - meaning it might as well be private within the wrapping class. Here you've got a public static inner class with mostly public methods, so functionally there'd be little to no change with it extracted.

Would different nodes have different flavors of required data? Would they fit well extending an abstract NodeRequiredData class, that has any utility value? Could Nodes with required data have an instance variable of type NodeRequiredData, initialized to the right flavor?